core/amber/src/main/python/core/runnables/data_processor.py

```diff
@@ -255,3 +258,16 @@ def _resume(self) -> None:
                 self.context.pause_manager.resume()
                 self.context.input_queue.enable_sub()
             self.context.state_manager.transit_to(WorkerState.RUNNING)
+
+    @staticmethod
+    def cast_tuple_to_match_schema(output_tuple, schema):
+        # TODO: move this into Tuple, after making Tuple aware of Schema
+
+        # right now only support casting ANY to binary.
+        import pickle
+        for field_name in output_tuple.get_field_names():
+            field_value = output_tuple[field_name]
+            field = schema.field_by_name(field_name)
+            field_type = field.type if field is not None else None
+            if field_type == pyarrow.binary():
+                output_tuple[field_name] = b'pickle    ' + pickle.dumps(field_value)
```

Comment:
Why add this "pickle" prefix? Is it for future extensibility? Will we use different libraries to do the serialization and deserialization?

Reply:
I think binary can have multiple format and use cases.
In Python most of objects can be "pickle"-ed and transferred as binary, this is the main use case here.  
Besides python objects, we could also transfer images, videos as binaries.

Reply:
got it